runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1859,15 +1980,49 @@ J9::TransformUtil::transformIndirectLoadChainImpl(TR::Compilation *comp, TR::Nod
       return false;
       }
 
-   // Dereference the chain starting from baseAddress and get the field address
-   void *fieldAddress = dereferenceStructPointerChain(baseAddress, baseExpression, isBaseStableArray, node, comp);
-   if (!fieldAddress)
+   // in non-jitserver mode, we need to hold a mutex until the end of the function to ensure
+   // fieldAddress holds
+   TR::VMAccessCriticalSection transformIndirectLoadChain(comp);
+   if (!transformIndirectLoadChain.hasVMAccess()
```

Comment:
This check can be removed because the critical section guarantees we have VM access.